Objects/typeobject.c

```diff
@@ -81,7 +81,7 @@ class object "PyObject *" "&PyBaseObject_Type"
 
 #define END_TYPE_DICT_LOCK() Py_END_CRITICAL_SECTION2()
 
-#ifndef NDEBUG
+#if !defined(NDEBUG) || defined(Py_Debug)
```

Comment:
The name of the macro is not `Py_Debug` but `Py_DEBUG` And ideally, I think we should instead use `Py_DEBUG` rather than `NDEBUG`.

Reply:
We had that before, and that caused problems for non-debug builds with `--with-assertions`. This issue was then addressed by https://github.com/python/cpython/pull/140133, which, however, introduced a new problem with debug-without-assertions builds.